src/index.js

```diff
@@ -1,16 +1,170 @@
 import React from "react";
 import ReactDOM from "react-dom";
+import * as CourseUtils from "./validateCourseUtils";
+import * as SemesterUtils from "./validateSemesterUtils";
+import * as CommonUtils from "./commonUtils";
+import { makeStyles } from "@material-ui/core/styles";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableRow from "@material-ui/core/TableRow";
+import Paper from "@material-ui/core/Paper";
 
 import "./styles.css";
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Hello CodeSandbox</h1>
-      <h2>Start editing to see some magic happen!</h2>
-    </div>
-  );
+let TABLEROWS = [];
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    width: "100%",
+    marginTop: theme.spacing(3),
+    overflowX: "auto"
+  },
+  table: {
+    minWidth: 650
+  }
+}));
+
+function createData(name, value) {
+  return { name, value };
+}
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    // this.state = { resultObj: {}, show: false };
+    this.state = { resultObj: { message: "", error: false } };
+    this.inputRef = React.createRef();
+    this.processInput = this.processInput.bind(this);
+    // this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  processInput() {
+    this.setState({
+      resultObj: this.validateCourseString(this.inputRef.current.value)
+    });
+  }
+
+  validateCourseString(str) {
+    let departmentName = "";
+    let courseNumber = "";
+    let semesterName = "";
+    let year = "";
+    let i = 0;
+
+    let resultFromCourseValidation = CourseUtils.validateCourse(
+      str,
+      i,
+      departmentName,
+      courseNumber
+    );
+
+    // console.log(resultFromCourseValidation);
+    if (resultFromCourseValidation.error === true) {
+      return resultFromCourseValidation;
+    } else {
+      str = resultFromCourseValidation.str;
+      courseNumber = resultFromCourseValidation.courseNumber;
+      departmentName = resultFromCourseValidation.departmentName;
+      i = resultFromCourseValidation.i;
+    }
+    // Check if space between Course and Semester Strings occur
+    if (i >= str.length) {
```

Comment:
return 0, here. In the parent function check if either of this.dname, this.cname, this.cnum, this.year are empty, then give error to user the type of information not found.

Ditto for other places.